docs/book/features/orphan_volumes.md

```diff
@@ -0,0 +1,50 @@
+## Orphan volumes detection and clean-up
+Orphan volumes are vSphere volumes that are present on a vSphere datastore but there is no corresponding PersistentVolume in Kubernetes clusters on the vCenter.
+
+Orphan volumes are often created when CNS solution creates more than one FCD for a Persistent Volume in the Kubernetes cluster. This could occur when the vCenter components are slow, storage is slow, vCenter service restarts, connectivity issues between vCenter and ESXi hosts etc. Since these orphan volumes occupy space in the datastore and are not really used in Kubernetes, it's useful to identify and cleanup orphan volumes periodically.
+
+This functionality provides a set of APIs to detect and delete orphan volumes on-demand, and also provide an option to turn on automatic deletion periodically.
+
+### What qualifies as an orphan volume ?
+A volume qualifies as an orphan volume if it meets all of the below conditions:
+
+1) A PersistentVolume in any of the registered kubernetes clusters is not using the volume.
+2) Volume was dynamically provisioned using vSphere CSI driver.
+3) Volume exists for more than 50 minutes(or `orphan-volume-detection-interval-mins` value configured in `cnsmanager-config` configmap) at the time of orphan volumes detection.
+
+### Which orphan volumes are skipped from detection/deletion ?
+Some volumes, even if a kubernetes PersistentVolume doesn't map to them, will not be considered during orphan volume detection/deletion. These include:
+* Volumes created out of band (not using vSphere CSI driver).
+* Statically provisioned CNS volume whose name doesn't start with `pvc-`.
+* File volumes.
+* Orphan volumes that contain snapshots will be detected as orphans but they can not be deleted using Orphan volume delete API.
+
+### A word of caution!
+Orphan volumes once deleted from a datastore cannot be recovered. It is therefore imperative that you [register all the Kubernetes clusters](../../../README.md#register-kubernetes-clusters-before-you-start) in vCenter with CNS Manager, so that an orphan volume is detected correctly and no crucial data loss happens when they're deleted.
```

Comment:
what if the Kubernetes cluster is registered and then the cluster itself is deleted without de-registering from CNS-Manager service?

Reply:
Can we document the requirement for de-registering such clusters for the proper functioning of overall orphan volume cleanup?

Reply:
>what if the Kubernetes cluster is registered and then the cluster itself is deleted without de-registering from CNS-Manager service?

If the cluster is deleted, then volumes will not be deregistered from CNS, right? There's one validation in orphan volume computation to see if the volume exists in CNS. If it does, those volumes are not qualified as orphans.

We can document about de-registering cluster before deleting it in the cluster registration section itself.

Reply:
I have added this in README in cluster registration section:
`If a registered cluster later gets decommissioned or deleted from the vCenter, don't forget to deregister it from CNS manager as well. This will ensure a smooth execution of functionalities offered through CNS manager.`